### ISSUE TITLE
Support for CSRF protection

### ISSUE THREAD

**Issue by stjohnjohnson** [silas/node-jenkins#28]
Jenkins added this feature a while back, but has now enabled it by default (and it's probably a good idea).

We're adding the csrf header in the instantiation of the Jenkins client for now, but it would be ideal to request new csrf automatically for each `post` request.

- https://wiki.jenkins-ci.org/display/JENKINS/Remote+access+API#RemoteaccessAPI-CSRFProtection
- https://wiki.jenkins-ci.org/display/JENKINS/CSRF+Protection

I'd be happy to provide a PR to add this, just let me know where it would be best to be implemented.

**Comment by silas** [comment]
This is released in `0.18.0` to npm. By default it's off, although once I have enough time to make it backwards compatible I'll enable it by default.

``` javascript
var jenkins = require('jenkins')({ baseUrl: 'http://user:pass@localhost:8080', crumbIssuer: true });
```

Also right now it makes a request for every POST since the expiration time isn't documented and I didn't dig into the code.

Let me know if you run into issues or have suggestions, and thanks for reporting the issue.

**Comment by stjohnjohnson** [comment]
👍 Awesome! Thanks! 
![woah](https://cloud.githubusercontent.com/assets/622065/17504367/2c7de60e-5dad-11e6-9158-bc796b69026a.gif)
